design/pipeline-resources.md

```diff
@@ -0,0 +1,190 @@
+    clean: boolean  # whether to fetch clean each time
+    fetchDepth: number  # the depth of commits to ask Git to fetch
+    lfs: boolean  # whether to download Git-LFS files
+    submodules: true | recursive  # set to 'true' for a single level of submodules or 'recursive' to get submodules of submodules
+    persistCredentials: boolean  # set to 'true' to leave the OAuth token in the Git config after the initial fetch
+```
+
+### Examples
+
+```yaml
+resources:         
+  repositories:
+  - repository: secondaryRepo      
+    type: GitHub
+    connection: myGitHubConnection
+    source: Microsoft/alphaworz
+```
+
+### `checkout` your repository
+
+Repos from the `repository` resources defined are automatically synced and made available for all the jobs in the pipeline. However, in any of the jobs, you can choose to override and sync only specific repository using `checkout` shortcut. 
+
+```yaml
+- checkout: string  # identifier for your repository; for primary repository use the keyword self.
+  clean: boolean  # whether to fetch clean each time
+  fetchDepth: number  # the depth of commits to ask Git to fetch
+  lfs: boolean  # whether to download Git-LFS files
+  submodules: true | recursive  # set to 'true' for a single level of submodules or 'recursive' to get submodules of submodules
+  persistCredentials: boolean  # set to 'true' to leave the OAuth token in the Git config after the initial fetch
+```
+
+### Example
+
+```yaml
+- checkout: secondaryRepo  
+  clean: false
+  fetchDepth: 5
+  lfs: true
+```
+
+When you use `checkout` to sync a specific repository resource, all the other repositories are not synced.  
+
+Or to avoid syncing any of the repository resources use:
+
+```yaml
+- checkout: none
+```
+
+Repositories are checked out to `$PIPELINES_RESOURCESDIR/<repository-identifier>/`.
+
+## Resources: `containers`
+
+If you need to consume a container image as part of your CI/CD pipeline, you can achieve it using `containers`. A container can be an Azure Container Registy or any external Docker registry.
+
+### Schema
+
+```yaml
+resources:          # types: pipelines | repositories | containers | packages
+  containers:
+  - container: string # identifier for the container resource      
+    type: enum # type of the registry like ACR, Docker etc. 
+    connection: string # service connection to connect to the image registry, defaults to ACR??
+    image: string # container image name, Tag/Digest is optional; defaults to latest image
+    options: string # arguments to pass to container at startup
+    env: { string:string } # list of environment variables to add
+```
+
+### Examples
+
+```yaml
+resources:         
+  containers:
+  - container: devLinux      
+    image: GitHub
+    connection: myDockerRegistry
+    source: Microsoft/alphaworz
+```
+
+Container image and metadata from the resource are made available across all the jobs in the pipeline for cosumption. 
```

Comment:
Can you clarify what this means? I think it's things like making variables available for `condition`s and the like.

Reply:
Sure, updated the info appropriately. Btw, we are currently working on the proposal for ACR and Helm specs. I will share the detailed proposal on containers along with triggers as a separate PR.